app/queries/follow_ups_query.rb

```diff
@@ -1,33 +1,27 @@
 class FollowUpsQuery
   def hypertension(region, period_type, group_by: nil)
-    query = Patient.joins(:blood_pressures)
-      .where("patients.recorded_at < #{BloodPressure.date_to_period_sql("blood_pressures.recorded_at", period_type)}")
-      .group_by_period(period_type, "blood_pressures.recorded_at", format: Period.formatter(period_type))
-      .distinct
-      .where(blood_pressures: {facility_id: region.facility_ids})
-      .with_hypertension
+    query = with(BloodPressure, period_type, at_region: region, format: Period.formatter(period_type)).with_hypertension
+
     if group_by.present?
-      results = query.group(group_by).count
-      sum_groups_per_period(results)
+      query.group(group_by).count.then { |results| sum_groups_per_period(results) }
     else
       query.count
     end
   end
 
-  def self.with(model_name, period, time_column: "recorded_at", at_region: nil, current: true, last: nil)
-    raise ArgumentError, "Only day, month and quarter allowed" unless period.in?([:day, :month, :quarter])
+  def with(model, period_type, time_column: "recorded_at", at_region: nil, **groupdate_opts)
```

Comment:
The key change here is accepting a blob of options to pass transparently to groupdate.